src/test/java/picard/sam/DownsampleSamTest.java

```diff
@@ -0,0 +1,158 @@
+package picard.sam;
+
+import htsjdk.samtools.*;
+import htsjdk.samtools.util.BufferedLineReader;
+import htsjdk.samtools.util.IOUtil;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import picard.cmdline.CommandLineProgramTest;
+import picard.sam.util.SamTestUtil;
+import htsjdk.samtools.DownsamplingIteratorFactory.Strategy;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+import static org.testng.Assert.assertEquals;
+import static picard.sam.PositionBasedDownsampleSamTest.assertGreaterThan;
+import static picard.sam.PositionBasedDownsampleSamTest.assertLessThan;
+
+/**
+ * Created by farjoun on 9/21/17.
+ */
+public class DownsampleSamTest extends CommandLineProgramTest {
+    private final static String sample = "TestSample";
+    private final static String readGroupId = "TestReadGroup";
+    private final static String platform = "ILLUMINA";
+    private final static String library = "TestLibrary";
+
+    private final static File TEST_DIR = new File("testdata/picard/sam/PositionalDownsampleSam/");
+    private final File dict = new File(TEST_DIR, "header.dict");
+
+    private File tempDir;
+    private File tempSamFile;
+
+    private SAMRecordSetBuilder setBuilder = new SAMRecordSetBuilder(true, SAMFileHeader.SortOrder.coordinate);
+    private SAMReadGroupRecord readGroupRecord = new SAMReadGroupRecord(readGroupId);
+
+    @Override
+    public String getCommandLineProgramName() {
+            return DownsampleSam.class.getSimpleName();
+    }
+
+    //create a samfile with one tile and randomly placed reads within it.
+    @BeforeTest
+    void setupBuilder() throws IOException {
+        final int numReads = 10000;
+        final String flowCellBarcode = "TESTBARCODE";
+        final int maxX = 10000;
+        final int maxY = 20000;
+        final int minX = 1000;
+        final int minY = 2000;
```

Comment:
Is there a way to share code between tests? Presumably DownsampleSam shouldn't actually take the X:Y coordinates into account so maybe this isn't necessary?